src/main/java/kr/codesqaud/cafe/repository/article/JdbcArticleRepository.java

```diff
@@ -1,49 +1,87 @@
     }
 
     @Override
-    public Article findById(final Long id) {
-        final String SQL = "SELECT id, title, writer, contents, createdAt FROM articles WHERE id = :id LIMIT 1";
-        return jdbcTemplate.queryForObject(SQL,
-                Map.of("id", id),
-                BeanPropertyRowMapper.newInstance(Article.class));
+    public Article findById(Long id) {
+        final String sql = "SELECT id, title, writer, contents, createdAt FROM articles WHERE id = :id AND deleted = false LIMIT 1";
+        return jdbcTemplate.queryForStream(sql, Map.of("id", id), articleRowMapper)
+                .findFirst()
+                .orElseThrow(ArticleNotFoundException::new);
     }
 
     @Override
     public List<Article> findAll() {
-        final String SQL = "SELECT id, title, writer, contents, createdAt FROM articles";
-        return jdbcTemplate.query(SQL, BeanPropertyRowMapper.newInstance(Article.class));
+        final String sql = "SELECT id, title, writer, contents, createdAt FROM articles WHERE deleted = false";
+        return jdbcTemplate.query(sql, articleRowMapper);
     }
 
     @Override
-    public boolean exists(final Long id) {
-        final String SQL = "SELECT id FROM articles WHERE id = :id LIMIT 1";
-        final Integer count = jdbcTemplate.queryForObject(SQL,
+    public boolean exist(Long id) {
+        final String sql = "SELECT count(*) FROM articles WHERE id = :id AND deleted = false LIMIT 1";
+        final Integer count = jdbcTemplate.queryForObject(sql,
                 Map.of("id", id),
                 Integer.class);
         return count > 0;
     }
+
+    @Override
+    public Article findWithSurroundingArticles(Long id) {
+        final String sql = "SELECT id, title, writer, contents, createdAt, deleted, previousId, nextId "
+                + " FROM (SELECT id, title, writer, contents, createdAt, deleted, "
+                + " LAG(id, 1, 0) OVER(ORDER BY id ASC) AS previousId, "
+                + " LEAD(id, 1, 0) OVER(ORDER BY id ASC) AS nextId "
+                + " FROM articles WHERE deleted = false) WHERE id = :id AND deleted = false";
+
+        return jdbcTemplate.queryForStream(sql, Map.of("id", id), articleRowMapper)
+                .findFirst()
+                .orElseThrow(ArticleNotFoundException::new);
+    }
+
+    @Override
+    public int edit(Article article) {
+        final String sql = "UPDATE articles SET title = :title, contents = :contents WHERE id = :id";
+        Map<String, Object> parameter = Map.of(
+                "id", article.getId(),
+                "title", article.getTitle(),
+                "contents", article.getContents());
+        return jdbcTemplate.update(sql, parameter);
+    }
+
+    @Override
+    public int delete(final Long id) {
+        final String sql = "UPDATE articles SET deleted = true WHERE id = :id";
+        Map<String, Object> parameter = Map.of(
+                "id", id);
+        return jdbcTemplate.update(sql, parameter);
```

Comment:
별도의 parameter 설정이 불필요하지 않나 싶네요.
```suggestion
        return jdbcTemplate.update(sql, Map.of("id", id));
```